python/pyspark/pandas/frame.py

```diff
@@ -4545,6 +4545,15 @@ def to_pandas_on_spark(self, index_col: Optional[Union[str, List[str]]] = None)
             )
             return DataFrame(internal)
 
+    # Keep to_koalas for backward compatibility for now.
+    def to_koalas(self, index_col: Optional[Union[str, List[str]]] = None) -> "DataFrame":
+        warnings.warn(
+            "DataFrame.to_koalas is deprecated as of DataFrame.to_pandas_on_spark. "
```

Comment:
Maybe: `DataFrame.to_koalas is deprecated. Use DataFrame.to_pandas_on_spark instead.`

Reply:
Looks better! Let me address it :-)